compiler/rustc_data_structures/src/profiling.rs

```diff
@@ -739,7 +771,21 @@ pub fn print_time_passes_entry(
     dur: Duration,
     start_rss: Option<usize>,
     end_rss: Option<usize>,
+    format: TimePassesFormat,
+    unique: bool,
 ) {
+    if format == TimePassesFormat::Json {
```

Comment:
It would be nice to have protection here against the addition of other formats. Either by adding `assert_eq!(format, TimePassesFormat::Text)` after the `return`, or by having a `match` here, which could fall through to the code below in the `Text` case.